policykit/policyengine/utils.py

```diff
@@ -38,31 +53,95 @@ def construct_authorize_install_url(request, integration, community=None):
     redirect_uri = f"{settings.SERVER_URL}/{integration}/install"
-    encoded_redirect_uri = quote(redirect_uri, safe='')
+    encoded_redirect_uri = quote(redirect_uri, safe="")
 
     # store state in user's session so we can validate it later
     state = "".join([str(random.randint(0, 9)) for i in range(8)])
-    request.session['community_install_state'] = state
+    request.session["community_install_state"] = state
 
     # if not specified, metagov will create a new community and pass back the slug
-    community_slug = community.metagov_slug if community else ''
+    community_slug = community.metagov_slug if community else ""
     url = f"{settings.METAGOV_URL}/auth/{integration}/authorize?type=app&community={community_slug}&redirect_uri={encoded_redirect_uri}&state={state}"
     logger.debug(url)
     return url
 
+
 def get_starterkits_info():
     """
     Get a list of all starter-kit names and descriptions.
     """
     starterkits = []
     cur_path = os.path.abspath(os.path.dirname(__file__))
-    dir_path = os.path.join(cur_path, f'../starterkits')
+    dir_path = os.path.join(cur_path, f"../starterkits")
     for kit_file in os.listdir(dir_path):
         kit_path = os.path.join(dir_path, kit_file)
         f = open(kit_path)
         data = json.loads(f.read())
-        starterkits.append({
-            'name': data['name'],
-            'description': data['description']
-        })
+        starterkits.append({"name": data["name"], "description": data["description"]})
     return starterkits
+
+
+def initialize_starterkit_inner(community, kit_data, creator_token=None):
+    from policyengine.models import Policy, CommunityRole, CommunityUser
+    from django.contrib.auth.models import Permission
+
+    # Initialize platform policies from starter kit
+    for policy in kit_data["platform_policies"]:
+        Policy.objects.create(**policy, kind=Policy.PLATFORM, community=community)
+
+    # Initialize constitution policies from starter kit
+    for policy in kit_data["constitution_policies"]:
+        Policy.objects.create(
+            **policy,
+            kind=Policy.CONSTITUTION,
+            community=community,
+        )
+
+    # Initialize roles from starter kit
+    for role in kit_data["roles"]:
+        r = CommunityRole.objects.create(
+            role_name=role["name"],
+            name=f"{community.platform}: {community.community_name}: {role['name']}",
+            description=role["description"],
+            community=community,
+        )
+
+        if role["is_base_role"]:
+            old_base_role = community.base_role
+            community.base_role = r
+            community.save()
+            old_base_role.delete()
+
+        # Add PolicyKit-related permissions
+        r.permissions.set(Permission.objects.filter(name__in=role["permissions"]))
+
+        # Add permissions for each PlatformAction
+        action_content_types = get_action_content_types(community.platform)
```

Comment:
@rickygv99 this allowed me to get rid of that static `permissions` field on CommunityPlatform that you added in the last PR. Instead of having to write everything out, we just assume that each subclass of PlatformAction defined in an app is a Content Type that can have view, propose, and execute permissions.